compiler/rustc_trait_selection/src/traits/util.rs

```diff
@@ -323,6 +323,17 @@ pub fn iterator_trait_ref_and_outputs<'tcx>(
     (trait_ref, sig.yield_ty)
 }
 
+pub fn fused_iterator_trait_ref_and_outputs<'tcx>(
```

Comment:
This is unnecessary -- fused iterators have no associated types and `FusedIterator` has no generic args.